packages/nodes-base/nodes/HttpRequest/v2/HttpRequestV2.node.ts

```diff
@@ -0,0 +1,594 @@
+				requestOptions.encoding = null;
+
+				if (options.bodyContentType !== 'raw') {
+					requestOptions.body = JSON.stringify(requestOptions.body);
+					if (requestOptions.headers === undefined) {
+						requestOptions.headers = {};
+					}
+					if (['POST', 'PUT', 'PATCH'].includes(requestMethod)) {
+						requestOptions.headers['Content-Type'] = 'application/json';
+					}
+				}
+			} else if (options.bodyContentType === 'raw') {
+				requestOptions.json = false;
+			} else {
+				requestOptions.json = true;
+			}
+
+			// Add Content Type if any are set
+			if (options.bodyContentCustomMimeType) {
+				if (requestOptions.headers === undefined) {
+					requestOptions.headers = {};
+				}
+				requestOptions.headers['Content-Type'] = options.bodyContentCustomMimeType;
+			}
+
+			// Add credentials if any are set
+			if (httpBasicAuth !== undefined) {
+				requestOptions.auth = {
+					user: httpBasicAuth.user as string,
+					pass: httpBasicAuth.password as string,
+				};
+			}
+			if (httpHeaderAuth !== undefined) {
+				requestOptions.headers![httpHeaderAuth.name as string] = httpHeaderAuth.value;
+			}
+			if (httpQueryAuth !== undefined) {
+				if (!requestOptions.qs) {
+					requestOptions.qs = {};
+				}
+				requestOptions.qs![httpQueryAuth.name as string] = httpQueryAuth.value;
+			}
+			if (httpDigestAuth !== undefined) {
+				requestOptions.auth = {
+					user: httpDigestAuth.user as string,
+					pass: httpDigestAuth.password as string,
+					sendImmediately: false,
+				};
+			}
+
+			if (requestOptions.headers!['accept'] === undefined) {
+				if (responseFormat === 'json') {
+					requestOptions.headers!['accept'] = 'application/json,text/*;q=0.99';
+				} else if (responseFormat === 'string') {
+					requestOptions.headers!['accept'] = 'application/json,text/html,application/xhtml+xml,application/xml,text/*;q=0.9, */*;q=0.1';
+				} else {
+					requestOptions.headers!['accept'] = 'application/json,text/html,application/xhtml+xml,application/xml,text/*;q=0.9, image/*;q=0.8, */*;q=0.7';
+				}
+			}
+
+			try {
+				let sendRequest: any = requestOptions; // tslint:disable-line:no-any
+				// Protect browser from sending large binary data
+				if (Buffer.isBuffer(sendRequest.body) && sendRequest.body.length > 250000) {
+					sendRequest = {
+						...requestOptions,
+						body: `Binary data got replaced with this text. Original was a Buffer with a size of ${requestOptions.body.length} byte.`,
+					};
+				}
+				this.sendMessageToUI(sendRequest);
+			} catch (e) { }
+
+			if (authenticateWith === 'genericAuth' || authenticateWith === 'none') {
+				if (oAuth1Api) {
+					requestPromises.push(
+						this.helpers.requestOAuth1.call(this, 'oAuth1Api', requestOptions),
+					);
+				} else if (oAuth2Api) {
+					requestPromises.push(
+						this.helpers.requestOAuth2.call(this, 'oAuth2Api', requestOptions, { tokenType: 'Bearer' }),
+					);
+				} else {
+					requestPromises.push(
+						this.helpers.request(requestOptions),
```

Comment:
Instead of caling `this.helpers.request` I think we should take the opportunity to migrate to the new `httpRequest` helper instead.

It is a breaking change anyway, so we can make more changes. Also if there is any operation currently not supported it would be a great opportunity to improve the new `httpRequest` helper to accomodate. WDYT?

Reply:
Switching `request` for `httpRequest` requires significant refactoring of request options.

Better to refactor now or after the feature is implemented?

Reply:
Hum I think that making this change now would be better but also I have second thoughts since the `httpRequest` helper is not being widely used for 100% of the use cases and therefore might be missing some functionality or buggy.

This is something that can be changed in the future without introducing a breaking change once we think the `httpRequest` helper has matured enough.